### ISSUE TITLE
Copy of a fortran array uses unnecessary transpose

### ISSUE THREAD

**Issue by EmilyBourne** [pyccel/pyccel#1257]
### To Reproduce
Provide code to reproduce the behavior:
```python
    a = np.array([[1,  2,  4,  8, 16],
             [1,  2,  4,  8, 16],
             [1,  2,  4,  8, 16],
             [1,  2,  4,  8, 16],
             [1,  2,  4,  8, 16]], order="F")
    b = np.array(a, order="F")
```

### Error details
Provide the generated code, or the error message:
```fortran
    allocate(a(0:4_i64, 0:4_i64))
    a = transpose(reshape([[1_i64, 2_i64, 4_i64, 8_i64, 16_i64], [1_i64, &
          2_i64, 4_i64, 8_i64, 16_i64], [1_i64, 2_i64, 4_i64, 8_i64, &
          16_i64], [1_i64, 2_i64, 4_i64, 8_i64, 16_i64], [1_i64, 2_i64, &
          4_i64, 8_i64, 16_i64]], [5_i64, 5_i64]))
    allocate(b(0:4_i64, 0:4_i64))
    b = transpose(a)

```

### Expected behavior
`a` should not be transposed to create `b`

### Language
Fortran

### Additional context
Identified in PR #1256

**Comment by EmilyBourne** [comment]
Blocked by #1258

**Comment by EmilyBourne** [comment]
Blocked by #1258 as tests can't be added

**Comment by EmilyBourne** [comment]
Duplicate of #1241
